tests/lib/Controller/DiscoPowerTest.php

```diff
@@ -52,6 +66,51 @@ public function testDiscoPowerNoDiscoParams(): void
         $r = $c->main($request);
     }
 
+    public function testDiscoPowerHasDiscoParams(): void
+    {
+        Configuration::setPreLoadedConfig($this->discoconfig, 'module_discopower.php');
+
+        $request = Request::create(
+            '/disco.php',
+            'GET',
+        );
+        $_GET = [
+            'entityID' => 'https://example.com/sp',
+            'return'=>'https://example.com/acs',
+            'returnIDParam' => 'idpentityid'
+        ];
+        $_SERVER['REQUEST_URI'] = '/disco.php';
+
+        $c = new Controller\DiscoPower();
+
+        $r = $c->main($request);
+        $this->assertInstanceOf(RunnableResponse::class, $r);
+        $this->assertTrue($r->isSuccessful());
+    }
+
+    public function testDiscoPowerReturnUrlDisallowed(): void
+    {
+        Configuration::setPreLoadedConfig($this->discoconfig, 'module_discopower.php');
+
+        $request = Request::create(
+            '/disco.php',
+            'GET',
+        );
+        $_GET = [
```

Comment:
Get-parameters can go inside the Request::create as the third parameter

Reply:
That does not work, because the request object is not passed to the parent constructor, but the parent (in SSP core) reads $_GET directly.

Reply:
Ah, my bad... We should try to fix that upstream though

Reply:
Sure, could fix that someday but not in scope for this feature.